src/main/java/de/zalando/tip/zalenium/proxy/DockerSeleniumStarterRemoteProxy.java

```diff
@@ -515,7 +515,21 @@ private int findFreePortInRange(int lowerBoundary, int upperBoundary) throws Ill
                 }
             }
         }
-        throw new IllegalStateException("Could not find a free TCP/IP port to use for docker-selenium");
+        return -1;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (!super.equals(obj)) {
+            return false;
+        }
+        DockerSeleniumStarterRemoteProxy fObj = (DockerSeleniumStarterRemoteProxy) obj;
+        return setupCompleted == fObj.isSetupCompleted();
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this method to simply inherit it. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1185)